tests/sentry/uptime/test_models.py

```diff
@@ -15,3 +19,56 @@ def test(self):
         self.create_project_uptime_subscription(uptime_subscription=other_sub, project=other_proj)
         assert get_active_monitor_count_for_org(self.organization) == 2
         assert get_active_monitor_count_for_org(other_org) == 1
+
+
+class UniqueMonitorTest(UptimeTestCase):
+    def test(self):
+        self.create_uptime_subscription(
+            url="https://santry.io",
+            interval_seconds=60,
+            method="GET",
+        )
+        with pytest.raises(IntegrityError), transaction.atomic(
```

Comment:
Using the transaction here helps us avoid the problem where the connection ends up in a bad state after a unique constraint violation